js/plugins/googleai/src/gemini.ts

```diff
@@ -1508,6 +1512,366 @@ function toGeminiFunctionModeEnum(
   }
 }
 
+/**
+ * Creates a v2 Gemini model action.
+ * This function extracts the core generation logic from defineGoogleAIModel
+ * and adapts it for the v2 plugin API.
+ */
+export function createGeminiModel({
```

Comment:
why are you doing it this way instead of `s/ai.defineModel/model/` in `defineGoogleAIModel`?

Reply:
see https://github.com/firebase/genkit/commit/4431c9cd830c71927ec99ee138e9689fe42f9916#diff-78fa2d7ca927b8ddea6c049eb2bce4f0866339d39288cf7f7d3665a2b5872ff4L193

Reply:
ah sorry, this was just a draft, that makes sense!